lua/cfc_gmod_scripts/anti_joinspam/sv_anti_joinspam.lua

```diff
@@ -4,7 +4,18 @@ local banDuration = 3
 hook.Add( "CheckPassword", "GMS_AntiJoinSpam", function( steamid64, ip, _, _, name )
     connections[steamid64] = ( connections[steamid64] or 0 ) + 1
 
-    if connections[steamid64] >= 8 then
+    timer.Simple( 3, function()
+        local connectionCount = connections[steamid64] or 0
+        connectionCount = math.max( connectionCount - 1, 0 )
+
+        if connectionCount == 0 then
+            connectionCount = nil
+        else
+            connections[steamid64] = connectionCount
+        end
+    end )
```

Comment:
Might be simpler?
The timer will keep restarting as they connect.

```suggestion
    timer.Create( "GMS_ConnectCooldown", 10, 1, function()
        connections[steamid64] = nil
    end )
```